sporkfish/searcher/searcher_config.py

```diff
@@ -11,16 +11,41 @@ class SearchMode(Enum):
 
 
 class SearcherConfig(Configurable):
-    """Configuration class for the searcher.
+    """
+    Configuration class for the searcher.
 
     :param max_depth: Maximum depth for the search (default: 5).
+                      Controls how deeply the search algorithm explores the game tree.
     :type max_depth: int
-    :param mode: Search mode (default: SearchMode.SINGLE_PROCESS).
-    :type mode: SearchMode
+    :param search_mode: Search mode (default: SearchMode.SINGLE_PROCESS).
```

Comment:
Nice catch! Sorry I always forget to update doc strings or comments after renaming symbols.

I should learn to ask chatgpt more